src/cart/crt_rpc.c

```diff
@@ -1245,6 +1245,17 @@ crt_req_send_immediately(struct crt_rpc_priv *rpc_priv)
 	}
 	D_ASSERT(rpc_priv->crp_hg_hdl != NULL);
 
+	if (D_LOG_ENABLED(DB_NET)) {
+		uint64_t hlc = crt_hlc_get();
+
+		if (hlc > rpc_priv->crp_create_hlc) {
+			uint64_t delay = crt_hlc2msec(hlc - rpc_priv->crp_create_hlc);
+
+			if (delay > 20)
+				RPC_TRACE(DB_NET, rpc_priv, "RPC send took %lu ms.\n", delay);
```

Comment:
or maybe set the crt_sent_hld there?

Reply:
This I'd like to see whole time which was spend after RPC was created and actually put into the wire. This time will include whole queue time. Expiration timer starts only when RPC is put in inflight queue. Here I print the whole send time.